cbor2/encoder.py

```diff
@@ -114,6 +114,31 @@ def encode_map(encoder, value):
         encoder.encode(val)
 
 
+def encode_sortable_key(val):
+    '''Takes a key and calculates the length of it's optimal byte representation'''
+    if val.__class__ == bytes:
+        return len(val), val
+    if val.__class__ == unicode:
+        x = val.encode('utf-8')
+        return len(x), x
+    if val.__class__ == int:
+        bytelen = len(hex(abs(val))[1:]) // 2
+        return bytelen, val
```

Comment:
This test is insufficient. For example, the number 24 can be represented with a single byte in CBOR while 25 cannot. I'm afraid the code will get a bit more complex.

Reply:
From the RFC: "Sorting is performed on the bytes of the representation of the key
      data items without paying attention to the 3/5 bit splitting for
      major types."